contracts/ERC1155M.sol

```diff
@@ -583,6 +674,14 @@ contract ERC1155M is IERC1155M, ERC1155Supply, ERC2981, Ownable2Step, Reentrancy
         if (start >= end) revert InvalidStartAndEndTimestamp();
     }
 
+    /**
+     * @dev Validates the timestamp is not expired.
+     */
+    function _assertValidTimestamp(uint64 timestamp) internal view {
+        if (timestamp < block.timestamp - _timestampExpirySeconds)
```

Comment:
I don't think block producers gaming the timestamp is a particular risk here but since we're cross chain am I correct that the timestamp was fast/convenient vs using block number and having to calculate a bespoke value based on each chains block time and we accepted the timestamp risk since that's low?

Reply:
Good call, we never thought about cross chain scenario. The timestamp check is just to ensure freshness of signature. The good thing is we don't have crosschain use case yet (we were having layerzero cross mint but we turned it down). We should revisit if we want to do cross mint again.